cmake/Utils/MachineTriple.cmake

```diff
@@ -0,0 +1,303 @@
+    if (abi STREQUAL "armeabi-v7a")
+      set(arch "armv7a")
+      set(base "linux-androideabi")
+    elseif (abi STREQUAL "arm64-v8a")
+      set(arch "aarch64")
+      set(base "linux-android")
+    elseif (abi STREQUAL "x86")
+      set(arch "i686")
+      set(base "linux-android")
+    elseif (abi STREQUAL "x86_64")
+      set(arch "x86_64")
+      set(base "linux-android")
+    elseif (abi STREQUAL "riscv64")
+      set(arch "riscv64")
+      set(base "linux-android")
+    else ()
+      # Fallback from processor if ABI isn't set
+      set(arch "${CMAKE_SYSTEM_PROCESSOR}")
+      string(TOLOWER "${arch}" arch)
+      if (arch MATCHES "armv7")
+        set(arch "armv7a")
+        set(base "linux-androideabi")
+      elseif (arch MATCHES "aarch64|arm64")
+        set(arch "aarch64")
+        set(base "linux-android")
+      elseif (arch MATCHES "x86_64|amd64")
+        set(arch "x86_64")
+        set(base "linux-android")
+      elseif (arch MATCHES "i[3-6]86|x86")
+        set(arch "i686")
+        set(base "linux-android")
+      elseif (arch MATCHES "riscv64")
+        set(arch "riscv64")
+        set(base "linux-android")
+      endif ()
+    endif ()
+
+    # Append API level if we can (e.g. aarch64-linux-android21)
+    set(api "")
+    if (DEFINED ANDROID_PLATFORM AND NOT "${ANDROID_PLATFORM}" STREQUAL "")
+      string(REGEX REPLACE "android-?" "" api "${ANDROID_PLATFORM}")
+    endif ()
+
+    if (arch STREQUAL "armv7a")
+      set(ret "${arch}-${base}")
+    else ()
+      set(ret "${arch}-${base}")
+    endif ()
+    if (api)
+      set(ret "${ret}${api}")
+    endif ()
+    set(${out_var}
+        "${ret}"
+        PARENT_SCOPE
+    )
+    return()
+  endif ()
+
+  # 3c) Apple iOS (device & simulator). Works for Xcode + toolchains.
+  if (APPLE AND (CMAKE_SYSTEM_NAME STREQUAL "iOS" OR CMAKE_OSX_SYSROOT MATCHES "[iI]phone"))
+    # Choose first arch if multi-arch is set
+    set(archs "${CMAKE_OSX_ARCHITECTURES}")
+    if (NOT archs)
+      set(archs "${CMAKE_SYSTEM_PROCESSOR}")
+    endif ()
+    list(GET archs 0 arch)
+    string(TOLOWER "${arch}" _arch_l)
+    if (_arch_l MATCHES "aarch64|arm64|arm64e") # iOS uses 'arm64' in triples
+      set(arch "arm64")
+    elseif (_arch_l MATCHES "x86_64|amd64")
+      set(arch "x86_64")
+    endif ()
+
+    # Simulator?
+    set(is_sim OFF)
+    if (CMAKE_OSX_SYSROOT MATCHES "simulator" OR CMAKE_XCODE_EFFECTIVE_PLATFORMS MATCHES
+                                                 "simulator"
+    )
+      set(is_sim ON)
+    endif ()
+
+    # Deployment target (best-effort)
+    set(ios_ver "")
+    foreach (maybe_ver CMAKE_OSX_DEPLOYMENT_TARGET CMAKE_IOS_DEPLOYMENT_TARGET
+                       IOS_DEPLOYMENT_TARGET
+    )
+      if (DEFINED ${maybe_ver} AND NOT "${${maybe_ver}}" STREQUAL "")
+        set(ios_ver "${${maybe_ver}}")
+        break()
+      endif ()
+    endforeach ()
+
+    if (is_sim)
+      set(ret "${arch}-apple-ios${ios_ver}-simulator")
```

Comment:
should this be `-simulator` or `-sim`? looks like `-simulator` exists: https://github.com/search?q=repo%3Allvm%2Fllvm-project+arm64-apple-ios-simulator&type=code, while many other projects such as `tokenizer-cpp` uses `-sim`